mypy/semanal.py

```diff
@@ -4437,7 +4411,7 @@ def named_type_or_none(self, qualified_name: str,
         if args is not None:
             # TODO: assert len(args) == len(node.defn.type_vars)
             return Instance(node, args)
-        return Instance(node, [AnyType(TypeOfAny.unannotated)] * len(node.defn.type_vars))
+        return Instance(node, [AnyType(TypeOfAny.special_form)] * len(node.defn.type_vars))
```

Comment:
It's quite confusing that `builtin_type` and `named_type` use `TypeOfAny.special_form` while `named_type_or_none` uses `TypeOfAny.unannotated`. Is it correct that changing it to `special_form`